src/Microsoft.Repl.Tests/Parsing/CoreParseResultTests.cs

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Linq;
+using Microsoft.Repl.Parsing;
+using Xunit;
+
+namespace Microsoft.Repl.Tests.Parsing
+{
+    public class CoreParseResultTests
+    {
+        [Fact]
+        public void Slice_WithZero_ReturnsThis()
+        {
+            CoreParser parser = new CoreParser();
+            string commandText = "set header content-type application/json";
+            ICoreParseResult parseResult = parser.Parse(commandText, commandText.Length);
+
+            ICoreParseResult result = parseResult.Slice(0);
+
+            Assert.Same(parseResult, result);
+        }
+
+        [Fact]
+        public void Slice_WithTooMany_ReturnsDefaultResult()
+        {
+            CoreParser parser = new CoreParser();
+            string commandText = "set header content-type application/json";
+            ICoreParseResult parseResult = parser.Parse(commandText, commandText.Length);
+
+            ICoreParseResult result = parseResult.Slice(100);
+
+            Assert.Equal(0, result.CaretPositionWithinCommandText);
+            Assert.Equal(0, result.CaretPositionWithinSelectedSection);
+            Assert.Equal(string.Empty, result.CommandText);
+            Assert.Single(result.Sections);
+            Assert.Contains(string.Empty, result.Sections);
+            Assert.Equal(0, result.SelectedSection);
+            Assert.Single(result.SectionStartLookup);
+            Assert.Equal(0, result.SectionStartLookup.Single().Key);
+            Assert.Equal(0, result.SectionStartLookup.Single().Value);
+        }
+
+        [Theory]
+        [InlineData("set header content-type application/json", 1, "header content-type application/json")]
+        [InlineData("set header content-type application/json", 2, "content-type application/json")]
+        [InlineData("set header content-type application/json", 3, "application/json")]
+        public void Slice_CorrectCommandText(string commandText, int toRemove, string expectedCommandText)
```

Comment:
nit: Can we follow this naming convention-<MethodBeingTested>_<DescribeInputScenario>_<DescribeExpectedResult>?

Reply:
I have a hard time following that when using the Theory and parameterized tests. The input scenario is a little harder to define. I gave it a shot in 4a073b8 though.